### DISCUSSION TITLE
Access directives in a custom rust plugin

### DISCUSSION THREAD

**Discussion by Meemaw** [apollographql/router#3494] [General]
Is it possible to somehow access directives on the client query/schema? I see directives are available in the router (on the schema ~ `schema.type_system.definitions.directives`), so its just the matter of passing them to the plugin init function?

**Comment by Meemaw** [comment]
@Geal @abernix any thoughts on this?

**Reply by Geal** [reply]
we are still in the process of redefining the internal code to manipulate schemas and queries using [ApolloCompiler](https://crates.io/crates/apollo-compiler), so if we gave access to that, we would risk some breaking changes later
